File: 128_longest_consecutive_sequence/longestConsecutive.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        n = set(nums)

        m = 0

        for x in n:
            if x - 1 in n:
                continue
            y = x + 1
            while y in n:
                y = y + 1
            m = max(m, y - x)

        return m

File: 128_longest_consecutive_sequence/longestConsecutive.go
package longestconsecutivesequence

func longestConsecutive(nums []int) int {
	set := make(map[int]struct{})

	for _, n := range nums {
		set[n] = struct{}{}
	}

	res := 0

	for x := range set {
		if _, ok := set[x-1]; ok {
			continue
		}

		y := x + 1

		for {
			if _, ok := set[y]; !ok {
				break
			}

			y += 1
		}

		res = max(res, y-x)
	}

	return res
}
